### PR TITLE
[Core/Questing] Fix quest objective loading

### PR DESCRIPTION
**Pull request by jasongdove**
This fixes a regression from https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/pull/262

1. Create new blood elf
2. Accept first quest
3. Kill 1x mana wyrm
4. Verify quest objective shows 1/6 kills
5. Log out
6. Log in to same character
7. Verify quest objective still shows 1/6 kills

This also removes some unused quest special flags and the associated pserver quests from the db.

### CODE REVIEW

File: src/server/game/Quests/QuestDef.h [hunk 1]
```diff
@@ -191,11 +191,7 @@ enum QuestSpecialFlags
     QUEST_SPECIAL_FLAGS_CAST                 = 0x020,   // Set by 32 in SpecialFlags in DB if the quest requires RequiredOrNpcGo killcredit but NOT kill (a spell cast)
     // room for more custom flags
 
-    // i-core-specific flags, going from the opposite direction
-    QUEST_SPECIAL_FLAGS_project_DAILY_QUEST = 0x01000000,   // Set by 16777216 in SpecialFlags in DB for some very specific quests
-    QUEST_SPECIAL_FLAGS_NO_REPUTATION_RATE   = 0x02000000,   // Set by 33554432 in SpecialFlags in DB if the quest reward should disregard server reputation gain rates
```
Comment (jasongdove) [inline src/server/game/Quests/QuestDef.h:196]: No quests have this flag in the DB, so it is not needed in the world sql update.

File: sql/updates/master/world/2024_07_18_00_world.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- QUEST_SPECIAL_FLAGS_project_DAILY_QUEST = 0x01000000
+
+delete
+    quest_template_addon,
+    quest_template
+from
+    quest_template_addon inner join quest_template on quest_template.ID = quest_template_addon.ID
+where
+    quest_template_addon.SpecialFlags & 0x01000000 > 0;
```
Comment (leelf00) [inline sql/updates/master/world/2024_07_18_00_world.sql:9]: Have you checked conditions table?

**Comment by leelf00** [comment]
quest_objective table data

**Comment by leelf00** [comment]
quest_offer_reward table

**Comment by leelf00** [comment]
quest_poi

File: sql/updates/master/world/2024_07_18_00_world.sql (hunk 2 above)
Comment (jasongdove) [reply sql/updates/master/world/2024_07_18_00_world.sql]: I don't see anything relevant in the conditions table based on this query

```sql
select *
from conditions
where ConditionTypeOrReference in (8, 9, 14, 28, 43, 47, 105, 106)
  and ConditionValue1 >= 40000
```

Comment (leelf00) [reply sql/updates/master/world/2024_07_18_00_world.sql]: My mistake

**Comment by leelf00** [comment]
I suggest use SQL condition like  

IN (40311,40310,40309,40308,40307,40306,40305,40304,40303,40302,40301,40300,40206,40205,40204,40203,40202,40201,40200,40193,40192,40191,40190,40155,40154,40153,40152,40151,40150,40149,40148,40147,40146,40145,40144,40143,40142,40141,40140,40139,40138,40137,40136,40135,40134,40133,40132,40131,40130,40129,40128,40127,40126,40125,40124,40123,40122,40121,40120,40115,40114,40113,40112,40111,40110,40109,40108,40107,40106,40105,40104,40103,40102,40101,40100,40040,40039,40038,40037,40036,40035,40034,40033,40032,40031,40030,40029,40028,40027,40026,40025,40024,40023,40022,40021,40020,40019,40018,40017,40016,40015,40014,40013,40012,40011,40010,40009,40008,40007,40006,40005,40004,40003,40002,40001,40000)

I am not sure quest_tempate have all quest of version MOP.

**Comment by jasongdove** [comment]
What about quest id >= 40000? https://www.wowhead.com/quest=40000/a-true-hunter

The tables have *a lot* of data from WoD and beyond.

**Comment by leelf00** [comment]
Yes

**Pull request merged**